crab/actions/visual_prompt_actions.py

```diff
@@ -0,0 +1,336 @@
+
+    TRANSFORMERS_ENABLE = True
+except ImportError:
+    TRANSFORMERS_ENABLE = False
+
+BoxType = tuple[int, int, int, int]
+
+
+def check_transformers_import():
+    if not TRANSFORMERS_ENABLE:
+        raise ImportError(
+            "Please install the required dependencies to use this function by running"
+            " `pip install crab-framework[client]`"
+        )
+
+
+def _calculate_iou(box1, box2):
+    xA = max(box1[0], box2[0])
+    yA = max(box1[1], box2[1])
+    xB = min(box1[2], box2[2])
+    yB = min(box1[3], box2[3])
+
+    interArea = max(0, xB - xA) * max(0, yB - yA)
+    box1Area = (box1[2] - box1[0]) * (box1[3] - box1[1])
+    box2Area = (box2[2] - box2[0]) * (box2[3] - box2[1])
+    unionArea = box1Area + box2Area - interArea
+    iou = interArea / unionArea
+
+    return iou
+
+
+def _calculate_center(box) -> tuple[int, int]:
+    return (box[0] + box[2]) / 2, (box[1] + box[3]) / 2
+
+
+def _remove_invalid_boxes(boxes_with_label, width, height):
+    boxes = [box[0] for box in boxes_with_label]
+    boxes_to_remove = set()
+    for idx, box in enumerate(boxes):
+        if box[0] < 0 or box[1] < 0 or box[2] > width or box[3] > height:
+            boxes_to_remove.add(idx)
+            continue
+        if box[0] >= box[2] or box[1] >= box[3]:
+            boxes_to_remove.add(idx)
+            continue
+
+    boxes_filt = [
+        box for idx, box in enumerate(boxes_with_label) if idx not in boxes_to_remove
+    ]
+    return boxes_filt
+
+
+def _filter_boxes_by_center(boxes_with_label, center_dis_thresh):
+    boxes = [box[0] for box in boxes_with_label]
+    boxes_to_remove = set()
+    for i in range(len(boxes)):
+        if i in boxes_to_remove:
+            continue
+        center_i = _calculate_center(boxes[i])
+        for j in range(i + 1, len(boxes)):
+            center_j = _calculate_center(boxes[j])
+            # fmt: off
+            center_close = ((center_i[0] - center_j[0]) ** 2 + 
+                            (center_i[1] - center_j[1]) ** 2 < 
+                            center_dis_thresh**2)
+            # fmt: on
+            if center_close:
+                boxes_to_remove.add(j)
+
+    boxes_filt = [
+        box for idx, box in enumerate(boxes_with_label) if idx not in boxes_to_remove
+    ]
+    return boxes_filt
+
+
+def _box_a_in_b(a: BoxType, b: BoxType):
+    return a[0] > b[0] and a[1] > b[1] and a[2] < b[2] and a[3] < b[3]
+
+
+def _filter_boxes_by_overlap(boxes_with_label):
+    boxes = [box[0] for box in boxes_with_label]
+    boxes_to_remove = set()
+    for i in range(len(boxes)):
+        if i in boxes_to_remove:
+            continue
+        for j in range(len(boxes)):
+            if _box_a_in_b(boxes[i], boxes[j]):
+                boxes_to_remove.add(j)
+
+    boxes_filt = [
+        box for idx, box in enumerate(boxes_with_label) if idx not in boxes_to_remove
+    ]
+    return boxes_filt
+
+
+def _filter_boxes_by_iou(
```

Comment:
Nit: may consider decouple the filtering mechanism from the actual filter policy

Reply:
May be not necessary currently